app/questionnaire/placeholder_parser.py

```diff
@@ -140,6 +170,53 @@ def _resolve_value_source_list(
                 values.append(value)
         return values
 
+    def _get_routing_path_block_ids(
+        self, sections_to_ignore: list | None = None
+    ) -> dict[str, list[str]]:
+        if self._location:
+            return get_block_ids_for_calculated_summary_dependencies(
+                schema=self._schema,
+                location=self._location,
+                progress_store=self._progress_store,
+                path=self._path_finder,
+                sections_to_ignore=sections_to_ignore,
+            )
+        return {}
+
     def _all_value_sources_metadata(self, placeholder: Mapping) -> bool:
         sources = self._schema.get_values_for_key(placeholder, key="source")
         return all(source == "metadata" for source in sources)
+
+
+def get_block_ids_for_calculated_summary_dependencies(
```

Comment:
There is an issue with this in that, every time placeholder render is called, we will evaluate the path, although we cache the path, if the renderer is different, it will reevaluate. If you look at section summary, you can see that the path is evaluated multiple times, this is because a renderer is used to resolve the title for the block which is different to the used for the summary blocks, but since we say the block has dependencies, it will evaluate the path even if no placeholder contain any calc summaries sources.

You will want to add a check to see if there is any calc summary sources. I would check if after finding out if you have any dependencies as it would be more performant as dependencies are pre resolved.

I would make a generic method like this that you can reuse.
```py
    def get_sources_for_type_from_data(self, *, source_type: str, data, ignore_keys):
        sources = self._get_mappings_with_key(
            "source", data, ignore_keys=ignore_keys
        )

        return [
            source
            for source in sources
            if source["source"] == source_type
        ]
```

Then it would be:

```py
if dependents and not calc_sources := get_sources_for_type_from_data("calculated_summary", data=data, ignore_keys=["when"]):
	return blocks_id_by_section
```

- `data` would be the placeholders that you need to pass down.

Reply:
Have updated this method and added the logic above 👍

Reply:
Moved this now 👍